python/examples/bare_matmul.py

```diff
@@ -0,0 +1,39 @@
+import torch
+
+import triton
+import triton.language as tl
+import benchmark
+
+
+@triton.jit
+def bare_matmul(X, Y, Z, BLOCK_SIZE: tl.constexpr):
+    pid = tl.program_id(0)
+    
+    offs_x = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    offs_y = tl.arange(0, BLOCK_SIZE)
+
+    x = tl.load(X + offs_x[:, None])
+    y = tl.load(Y + offs_y[None, :])
+
+    z = tl.dot(x, y)
+    tl.store(Z + offs_x[:, None] + offs_y[None, :], z)
+
+
+@benchmark.measure()
+def bench_matmul(M, N, K, provider):
+    device = 'cpu'
+    dtype = torch.float32
+    a = torch.randn((M, K), device=device, dtype=dtype)
+    b = torch.randn((K, N), device=device, dtype=dtype)
+    c = torch.empty((K, N), device=device, dtype=dtype)
+    if provider == 'torch':
+        torch.matmul(a, b)
+    if provider == 'triton':
+        bare_matmul[(1,)](a, b, c, N)
+
```

Comment:
i think it would be useful to compare the results between torch and triton

Reply:
great idea, I added "test" provider which makes comparisons